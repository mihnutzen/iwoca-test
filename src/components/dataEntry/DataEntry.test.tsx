import React from 'react';
import { shallow } from 'enzyme';
import DataEntry from './DataEntry';

describe('DataEntry component', () => {
  it('Should render correctly', () => {
    const props = {
      label: 'label',
      value: 100,
      unit: '£',
      onChnage: jest.fn(),
      error: 'error'
    };
    const wrapper = shallow(<DataEntry {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the onchange function', () => {
    const changeFn = jest.fn();
    const props = {
      label: 'label',
      unit: '£',
      onChnage: changeFn
    };
    const wrapper = shallow(<DataEntry {...props} />);
    wrapper.find('input').simulate('change', { target: { value: 100 } });

    expect(changeFn).toHaveBeenCalledWith(100);
  });
});
