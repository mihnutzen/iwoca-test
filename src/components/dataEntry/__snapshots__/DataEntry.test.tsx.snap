// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataEntry component Should render correctly 1`] = `
<div
  className="DataEntry"
>
  <span>
    label
  </span>
  <span>
    <input
      max={1000000}
      min={1}
      onChange={[Function]}
      type="number"
      value={100}
    />
  </span>
  <span>
    (in 
    Â£
    )
  </span>
  <span
    className="DataEntry__error"
  >
    error
  </span>
</div>
`;
